### PR TITLE
fix caching for CompressJarTask

### PR DESCRIPTION
**Pull request by rhysdh540**
`shadowJar` outputs to a different place, which allows caching to work 👍

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -101,13 +101,15 @@ val patchHistory = releaseTags
...
 
 val maxPatch = patchHistory.maxOfOrNull { it.substringBefore('-').toInt() }
-val patch = 
-	maxPatch?.plus(
-		if (patchHistory.contains(maxPatch.toString()))
-			releaseIncrement
-		else
-			0
-	) ?: 0
+val patch = maxPatch.let {
```
Comment (Nolij) [inline build.gradle.kts:104]: Why?

File: buildSrc/src/main/kotlin/dev/nolij/zumegradle/JarCompressing.kt [hunk 2]
```diff
@@ -306,10 +308,18 @@ open class CompressJarTask : DefaultTask() {
 
 	@TaskAction
 	fun compressJar() {
-		if (useProguard)
-			applyProguard(inputJar, minecraftConfigs, project.rootDir)
-		squishJar(inputJar, jsonShrinkingType, mappingsFile)
-		deflate(outputJar, deflateAlgorithm)
+//		if (useProguard)
```
Comment (Nolij) [inline buildSrc/src/main/kotlin/dev/nolij/zumegradle/JarCompressing.kt:311]: Any particular reason we're keeping these comments around?

**Review by Nolij — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/dev/nolij/zumegradle/JarCompressing.kt (hunk 2 above)
Comment (rhysdh540) [reply buildSrc/src/main/kotlin/dev/nolij/zumegradle/JarCompressing.kt]: nope

File: build.gradle.kts (hunk 1 above)
Comment (rhysdh540) [reply build.gradle.kts:104]: readability mostly

Comment (Nolij) [reply build.gradle.kts:104]: This `.let` syntax bothers me for reasons I can't explain. Can you revert this and I'll just clean it up myself?

**Review by Nolij — APPROVED** [review]

**Pull request merged**